Add "garden map" (technical roadmap) content (#10)
software-gardening/almanac: src/almanac/garden-circle/garden-map.md

# Garden Map

Our garden map is where you can learn about what features we're working on, what stage they're in, and when we expect to bring them to you.
The content here roughly follows that of a [technical roadmap](https://en.wikipedia.org/wiki/Technology_roadmap) (providing strategic visibility on implementation details and timelines).

## We'd love your input

We welcome your input on the garden map and elements found within it!
Please see the [contributing](contributing.md) guide for more information on how to participate in this process.

## Active Garden Map(s)

The following are active garden maps for the Software Gardening Almanac.
Please see below for a further description of how these projects are organized and used.

<!--alex disable fellowship -->

> - [__2024 BSSw Fellowship Milestones__](https://github.com/orgs/software-gardening/projects/1): this project is used to help communicate activities associated with the 2024 [BSSw Fellowship](https://bssw.io/pages/bssw-fellowship-program) associated with the Software Gardening Almanac.

Reviewer: ```suggestion
- [__2024 BSSw Fellowship Milestones__](https://github.com/orgs/software-gardening/projects/1): this project communicates activities associated with the 2024 [BSSw Fellowship](https://bssw.io/pages/bssw-fellowship-program) associated with the Software Gardening Almanac.
```

Simplify and convert to active voice

Author: Thanks, will add this in!

<!--alex enable fellowship -->

> ## Sections

Reviewer: In the future (maybe we don't have enough content yet), it might be nice to add actual links to concrete sections within each view.

Author: This is a great suggestion, I'll add this as a new issue for later work. I agree that, for the time being, we don't yet have enough content to meaningfully differentiate (this will likely change).

Author: Opened #11 to address this at a later point.

> The garden map can be understood through the following sections.

Reviewer: ```suggestion
You can explore the garden map with different views:
```

Switching to active voice and clarify

Author: Thank you, a lot more clear!

> - __"Canopy-view" development visibility__: new, existing, and previous work is tracked at a high-level using organization-wide [GitHub Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

Reviewer: ```suggestion
- __"Canopy-view" development visibility__: We track new, existing, and previous work at a high-level using organization-wide [GitHub Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).
```

> - __"Ground-level" development visibility__: work within GitHub Projects is tracked using repository-specific [GitHub issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and a field within GitHub Projects called "status" associated with each issue (indicated as "Todo", "In progress", "Paused", or "Done").

Reviewer: ```suggestion
- __"Ground-level" development visibility__: We track repository-specific work using [GitHub issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and a GitHub Project field called "status", which we associate with each issue (indicated as "Todo", "In progress", "Paused", or "Done").
```

> - __Major milestones__: major milestones are tracked using common [GitHub issues labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels) across all repositories through their associated issues.

Reviewer: ```suggestion
- __Major milestones__: We track major milestones using common [GitHub issues labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels) across all repositories through their associated issues.
```

## Visualizations

It can be helpful to visualize the garden map using data plots or other images.
We use [GitHub Projects insights](https://docs.github.com/en/issues/planning-and-tracking-with-projects/viewing-insights-from-your-project/about-insights-for-projects) to assist with automated creation and updates to garden map activity.
Please see this [project insights page](https://github.com/orgs/software-gardening/projects/1/insights) for an example of these (using the insights button within any GitHub Projects for viewing others).

> ## Ethos

Reviewer: This feels like it might be mentioned earlier in the map

Author: Good point, thank you! I'll move this to earlier in the document.

> The garden map is intended to provide useful tools for reaching Software Gardening Almanac realizations.

Reviewer: ```suggestion
We intend for the garden map to provide useful tools for us to efficiently create and maintain the Software Gardening Almanac.
```

Clarify and simplify through slight elaboration

Author: Thanks! Lots better.
